jupyter_theme_editor/handlers.py

```diff
@@ -0,0 +1,42 @@
+import json
+from jupyter_server.base.handlers import APIHandler
+from jupyter_server.utils import url_path_join
+import tornado
+from jinja2 import Environment, PackageLoader
+
+class RouteHandler(APIHandler):
+    # The following decorator should be present on all verb methods (head, get, post,
+    # patch, put, delete, options) to ensure only authorized user can request the
+    # Jupyter server
+
+    @tornado.web.authenticated
+    def post(self):
+        # input_data is a dictionary with a key "name"
+        input_data = self.get_json_body()
+        new_input_data = {}
+        for key, value in input_data.items():
+            value = str(input_data[key])
+            new_value = value.strip()
+            new_key = key.replace('--jp-', '').replace('-', '_')
+            new_input_data[new_key] = new_value
+
+        env = Environment(loader=PackageLoader(
+            "jupyter_theme_editor", "templates"))
+        j2_template = env.get_template("template.css")
```

Comment:
We can do thing a bit better here because the `Environment` and finding the template is gonna be always the same. We can pass them in the `initialize` to avoid recreating them every time.

See that example:

https://github.com/jupyterlab/pull-requests/blob/453ef305f3b0fe4d1781aa2ecea85d7a2ec9c3f0/jupyterlab_pullrequests/handlers.py#L32

And how to pass the keyword arguments to initialize

https://github.com/jupyterlab/pull-requests/blob/453ef305f3b0fe4d1781aa2ecea85d7a2ec9c3f0/jupyterlab_pullrequests/handlers.py#L226

Reply:
A tentative changed is proposed in commit [c8c9a6e](https://github.com/jupyterlab-contrib/jupyterlab-theme-editor/pull/7/commits/c8c9a6ee231ce9bde36918cd0eb321c41343bd83)